prepare_run.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+import subprocess
+import argparse
+
+
+# Usage: ./prepare_run.py NAO-ONT-20240519-practice
+# This script batches pod5 files from a delivery directory and prepares them for processing
+
+def list_s3_files(S3_BUCKET, delivery):
+    """List files in S3 with the given prefix"""
+    pod5_dir = f"{delivery}/pod5/"
+    cmd = ["aws", "s3", "ls", f"s3://{S3_BUCKET}/{pod5_dir}"]
+
+    result = subprocess.run(cmd, stdout=subprocess.PIPE, text=True, check=True)
+
+    files = []
+    for line in result.stdout.strip().split('\n'):
+        if not line:
+            continue
+        parts = line.split()
+        if len(parts) >= 4:  # Standard S3 ls output format
+            file_name = ' '.join(parts[3:])
+            size = int(parts[2])
+            files.append((file_name, size))
+
+    return files
+
+
+def copy_file_to_s3(S3_BUCKET, pod5_file, delivery, batch_id):
```

Comment:
This should just take a pair of paths (origin and destination) and the logic of figuring out those paths should be outside the function.  It divides the work more cleanly.